lib/AST/ModuleDependencies.cpp

```diff
@@ -536,19 +536,14 @@ swift::dependencies::registerCxxInteropLibraries(
                     })) {
     // Only link with CxxStdlib on platforms where the overlay is available.
     switch (Target.getOS()) {
-    case llvm::Triple::Linux:
-      if (!Target.isAndroid())
-        RegistrationCallback(LinkLibrary("swiftCxxStdlib",
-                                         LibraryKind::Library));
-      break;
     case llvm::Triple::Win32: {
       RegistrationCallback(
           LinkLibrary(hasStaticCxxStdlib ? "libswiftCxxStdlib" : "swiftCxxStdlib",
                       LibraryKind::Library));
       break;
     }
     default:
-      if (Target.isOSDarwin())
+      if (Target.isOSDarwin() || Target.isOSLinux())
```

Comment:
I guess that BSD is still out otherwise it would be nice to collapse this switch.